pkg/daemon/config.go

```diff
@@ -122,20 +122,44 @@ func ParseFlags() (*Configuration, error) {
 }
 
 func (config *Configuration) initNicConfig() error {
+	var (
+		iface   *net.Interface
+		err     error
+		encapIP string
+	)
+
 	if config.Iface == "" {
 		i, err := getDefaultGatewayIface()
 		if err != nil {
 			return err
 		} else {
-			config.Iface = i
+			iface, err = findInterface(i)
```

Comment:
the `config.Iface` here should be the interface that owns the `POD_IP`, not the default gateway

Reply:
The default gateway only for mtu, must use the interface that owns the POD_IP?

Reply:
yes, the packet will be tunneled by the interface that owns the encapIP, so the mtu should also coordinate with the same interface